xds/src/main/java/io/grpc/xds/ClientLoadCounter.java

```diff
@@ -72,6 +85,13 @@ void recordCallFinished(Status status) {
     }
   }
 
+  @Override
+  void recordMetric(String name, double value) {
+    MetricRecorder recorder =
+        metricRecorders[(int) (Thread.currentThread().getId() % THREAD_BALANCING_FACTOR)];
```

Comment:
Is this documented in some spec?

Reply:
No, this is a dirty trick we came up with to deal with atomic update of two items while keeping thread contention not too severe. But, it's an implementation detail. :smirk:

Reply:
Casting a long value to int could overflow and get a negative result.
```
System.out.println((int) 2147483700l  % 64);
```

```
-12
```

Reply:
The number to be casted will be less than `THREAD_BALANCING_FACTOR` after the modulo operation (more precisely, it's remainder operation). So overflow should not happen.

Reply:
Oh, I overlooked there is a bracket surrounding modulo expression. Is `Thread.getId()` never negative?

Reply:
Yes.
> [The thread ID is a positive long number generated when this thread was created.](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#getId())